engine/pkg/rpcerror/grpc.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpcerror
+
+import (
+	gerrors "errors"
+	"fmt"
+	"reflect"
+
+	"github.com/gogo/status"
+	"github.com/pingcap/errors"
+	"google.golang.org/grpc/codes"
+
+	pb "github.com/pingcap/tiflow/engine/enginepb"
+)
+
+func tryUnwrapNormalizedError(errIn error) (typeErasedNormalizedError, bool) {
```

Comment:
What does `typeErasedNormalizedError ` means literally?

Reply:
I get the scene and the point of `type erasure`.
Are we face the same situation? 
It's not so easy to get the meaning of `typeErasedNormalizedError` and what it does.

Reply:
How do you implement `Is` and `Convert`?